no_std compilation broken in v0.25.0

## Problem

v0.25.0 fails to compile without default features (no_std mode):

```
error[E0412]: cannot find type `String` in `builder.rs:126,416`
error[E0412]: cannot find type `Box` in `introspection.rs:238`
error[E0599]: no method `to_string` in `display.rs:391`
```

## Root Cause

Missing `alloc` imports when `std` feature is disabled.

## Solution

1. Add `use alloc::string::String;` to builder.rs
2. Add `use alloc::boxed::Box;` to introspection.rs
3. Add `use alloc::string::ToString;` to display.rs
4. Add CI check: `cargo check --no-default-features`

## Severity

**Critical** - Breaks no_std builds

## Hotfix

Will release v0.25.1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -25,8 +25,28 @@ jobs:
       all-features: true
     secrets: inherit
 
+  no-std:
+    name: no_std compatibility
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v5
+
+      - name: Install Rust (stable)
+        uses: dtolnay/rust-toolchain@v1
+        with:
+          toolchain: stable
+
+      - name: Cache cargo
+        uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: no-std
+          save-if: ${{ github.ref == 'refs/heads/main' }}
+
+      - name: Check no_std build
+        run: cargo check --no-default-features
+
   update-changelog:
-    needs: ci
+    needs: [ci, no-std]
     uses: ./.github/workflows/reusable-changelog.yml
     secrets:
       GH_TOKEN: ${{ secrets.GH_TOKEN }}

--- a/src/app_error/core/builder.rs
+++ b/src/app_error/core/builder.rs
@@ -2,7 +2,7 @@
 //
 // SPDX-License-Identifier: MIT
 
-use alloc::{borrow::Cow, sync::Arc};
+use alloc::{borrow::Cow, string::String, sync::Arc};
 use core::error::Error as CoreError;
 #[cfg(feature = "backtrace")]
 use std::backtrace::Backtrace;

--- a/src/app_error/core/display.rs
+++ b/src/app_error/core/display.rs
@@ -2,6 +2,7 @@
 //
 // SPDX-License-Identifier: MIT
 
+use alloc::string::ToString;
 use core::{
     error::Error as CoreError,
     fmt::{Formatter, Result as FmtResult},

--- a/src/app_error/core/introspection.rs
+++ b/src/app_error/core/introspection.rs
@@ -2,7 +2,7 @@
 //
 // SPDX-License-Identifier: MIT
 
-use alloc::borrow::Cow;
+use alloc::{borrow::Cow, boxed::Box};
 use core::error::Error as CoreError;
 
 #[cfg(feature = "backtrace")]

